lib/puppet/provider/vcenter.rb

```diff
@@ -66,7 +66,30 @@ def return_folder(folder)
     end
   end
 
-  def findvm(folder)
+  def findvm(folder,vm_name)
+    folder.children.each do |f|
+      break if @vm_obj
+      case f
+      when RbVmomi::VIM::Folder
+        findvm(f,vm_name)
+      when RbVmomi::VIM::VirtualMachine
+        @vm_obj = f if f.name == vm_name
+      when RbVmomi::VIM::VirtualApp
+        f.vm.each do |v|
+          if v.name == vm_name
+            @vm_obj = f
+            break
+          end
+        end
+      else
+        puts "unknown child type found: #{f.class}"
+        exit
```

Comment:
I believe this should raise an exception including the message, rather than write a string to standard ouput then exit :-)